www/js/lib/uiUtil.js

```diff
@@ -43,34 +43,87 @@ define(rqDef, function(settingsStore) {
+                    webpMachine.constructor.replaceImageWithCanvas(node, canvas);
+                } else {
+                    // For standard browsers that have access to the canvas, we simply convert the canvas to a data URI
+                    node.setAttribute(nodeAttribute, canvas.toDataURL());
+                }
             }).catch(function (err) {
                 console.error('There was an error decoding image in WebpMachine', err);
                 if (callback) callback();
             });
         } else {
-            var blob = new Blob([content], {
-                type: mimeType
-            });
-            var url = URL.createObjectURL(blob);
-            if (callback) callback(url);
-            node.addEventListener('load', function () {
-                URL.revokeObjectURL(url);
-            });
-            node.setAttribute(nodeAttribute, url);
+            if (callback) callback(); // Calling back as soon as possible speeds up extraction
+            // In browsers that support WebP natively, or for non-WebP images, we can simply convert the Uint8Array to a data URI
+            // DEV: we use FileReader method because btoa fails on utf8 strings (in SVGs, for example)
+            // See https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding#The_Unicode_Problem
+            // This native browser method is very fast: see https://stackoverflow.com/a/66046176/9727685
+            var myReader = new FileReader();
+            myReader.onloadend = function () {
+                var url = myReader.result;
+                node.setAttribute(nodeAttribute, url);
+            };
+            myReader.readAsDataURL(new Blob([content], { type: mimeType }));
+
+            // Below is BLOB URL-based method; slower and requires callback on every image load to revoke the BLOB
+            // var imgBlob = new Blob([content], {
+            //     type: mimeType
+            // });
+            // var url = URL.createObjectURL(imgBlob);
+            // if (callback) callback(url);
+            // node.addEventListener('load', function () {
+            //     URL.revokeObjectURL(url);
+            // });
+            // node.setAttribute(nodeAttribute, url);
+        }
+    }
+
+    /**
+     * Determines whether the Canvas Elements Workaround for decoding WebP images is needed, and sets UI accordingly
+     * Note that the workaround will never be used in Service Worker mode, but we still need to determine it in case the user switches to jQuery mode
+     * @returns {Boolean} A value to indicate whether the workaround may needed 
+     */
+    function determineCanvasElementsWorkaround() {
+        var userPreference = settingsStore.getItem('useCanvasElements') !== 'false';
+        // Determines setting automatically
+        var useCanvasElements = webpMachine && webpHero && !webpHero.detectCanvasReadingSupport();
```

Comment:
This variable name looks misleading to me: it stores the capability of the browser